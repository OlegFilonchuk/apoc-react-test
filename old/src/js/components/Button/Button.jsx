import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { upperFirst } from 'lodash';
import { getDataTestElementProps, dataTestElementPropTypes } from '../../utils/dataTestElementPropUtils';

import './Button.less';

const noop = () => null;

function Button(props) {
  const { className, isSelected, isLinkView, disabled, id, name, type, children, onClick, title, value } = props;

  const buttonClass = classNames('sc-btn', className, {
    'sc-on': isSelected,
    'sc-disabled': disabled,
    'link-view': isLinkView
  });

  const clickHandler = disabled ? noop : onClick;

  const processDataTestElement = dataTestElementValue =>
    `${dataTestElementValue}${value && typeof value === 'string' ? '-' : ''}${
      value && typeof value === 'string' ? upperFirst(value) : ''
    }`;

  return (
    <button
      id={id}
      name={name}
      type={type}
      disabled={disabled}
      className={buttonClass}
      onClick={clickHandler}
      title={title}
      {...getDataTestElementProps(props, processDataTestElement)}
    >
      {children}
    </button>
  );
}

Button.propTypes = {
  /**
   * Button id
   */
  id: PropTypes.string,

  /**
   * Is button disabled
   */
  disabled: PropTypes.bool,

  /**
   * Button name
   */
  name: PropTypes.string,

  /**
   * Button type
   */
  type: PropTypes.string,

  /**
   * Is button selected
   */
  isSelected: PropTypes.bool,

  /**
   * Is button looks like link
   */
  isLinkView: PropTypes.bool,

  /**
   * Is called when button is clicked
   */
  onClick: PropTypes.func.isRequired,

  /**
   * Inherited classes. For more details check Buttons section in Chameleon.
   * Defaults to 'sc-btn-default'.
   */
  className: PropTypes.string,

  /**
   * Button content
   */
  children: PropTypes.node.isRequired,

  /* eslint-disable react/no-unused-prop-types */
  /**
   * Button value that is used in ButtonGroup component
   */
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.object]),
  /* eslint-enable react/no-unused-prop-types */

  title: PropTypes.string,

  ...dataTestElementPropTypes
};

Button.defaultProps = {
  id: '',
  name: '',
  type: 'button',
  disabled: false,
  isSelected: false,
  isLinkView: false,
  value: null,
  className: 'sc-btn-default',
  'data-test-element': '',
  title: ''
};

export default Button;
