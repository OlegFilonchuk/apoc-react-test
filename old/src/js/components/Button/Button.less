@import '~chameleon/less/variables.less';

.sc-btn {
  cursor: pointer;

  &.link-view {
    color: @primary;
    background: none;
    padding: initial;

    &.sc-disabled {
      color: @disable;
    }
  }
}
