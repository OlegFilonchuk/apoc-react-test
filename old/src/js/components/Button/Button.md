## Basic button

Simple usage of the **Button** component:

    const callbackFn = () => {
        document.getElementById('clicked').textContent = 'I was clicked!';
    };

    (
      <div>
        <Button className="sc-btn-default" onClick={ callbackFn }>Push Me</Button>
        <div id="clicked">
            Please click button!
        </div>
      </div>
    )

## Disabled button

Rendering Button with a disables state

    <Button onClick={() => true} disabled={true}>You can't click me</Button>

## Colors

Button component can support different colors:

    const onClick = (text) => {
      console.log(`${text} clicked!`);
    };

    (
      <div>
        <p>
          <Button onClick={() => { onClick('Button normal'); }}>Button Normal</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Button Primary'); }}  className="sc-btn-primary">Button Primary</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Primary Outline'); }}  className="sc-btn-primary-outline">Button Primary Outline</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Button Warning'); }}  className="sc-btn-warning">Button Warning</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Button Error'); }}  className="sc-btn-error">Button Error</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Error Outline'); }}  className="sc-btn-error-outline">Button Error Outline</Button>
        </p>
        <p>
          <Button onClick={() => { onClick('Button Success'); }}  className="sc-btn-success">
            <i className="sc-icon-download" /> Button Success
          </Button>
        </p>
        <p>
          <Button isLinkView onClick={() => { onClick('Link Button'); }}  className="sc-primary">
            <i className="sc-icon-file" /> Link Button
          </Button>
        </p>
      </div>
    )
