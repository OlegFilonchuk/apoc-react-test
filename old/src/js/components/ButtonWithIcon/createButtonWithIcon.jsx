import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './ButtonWithIcon.less';

function createButtonWithIcon(Button) {
  function ButtonWithIcon({ className, iconClassName, children, adjustLineHeight, ...props }) {
    const finalClassName = classNames('button-with-icon', className);
    const finalIconClassName = classNames('button-with-icon__icon', iconClassName);
    const adjustLineHeightInPixels = adjustLineHeight ? `${adjustLineHeight} px` : adjustLineHeight;

    return (
      <Button {...props} className={finalClassName} style={{ lineHeight: adjustLineHeightInPixels }}>
        <i className={finalIconClassName} style={{ height: adjustLineHeightInPixels }} />
        {children && <span className={'button-with-icon__text'}>{children}</span>}
      </Button>
    );
  }

  ButtonWithIcon.propTypes = {
    className: PropTypes.string,
    children: PropTypes.node,
    iconClassName: PropTypes.string.isRequired,
    onClick: PropTypes.func,
    /**
     * This should equal to total height if an icon in pixels, to best vertical center alignment
     */
    adjustLineHeight: PropTypes.number
  };

  ButtonWithIcon.defaultProps = {
    adjustLineHeight: undefined,
    children: undefined,
    className: '',
    onClick: () => false
  };

  return ButtonWithIcon;
}

export default createButtonWithIcon;
