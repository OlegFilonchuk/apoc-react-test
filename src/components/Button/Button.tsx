import React from 'react';

export type ButtonProps = {
  label: string;
};

export function Button(props: ButtonProps) {
  return <button>{props.label}</button>;
}
