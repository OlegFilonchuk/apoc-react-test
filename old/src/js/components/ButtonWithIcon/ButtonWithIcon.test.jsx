import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';

import createButtonWithIcon from './createButtonWithIcon';
import Button from '../Button/Button';

describe('createButtonWithIcon', () => {
  const ButtonWithCustomIcon = createButtonWithIcon(Button);

  describe('with default set of props', () => {
    let component;
    let errorConsole;

    beforeAll(() => {
      errorConsole = spyOn(console, 'error');
    });

    beforeEach(() => {
      component = mount(
        <ButtonWithCustomIcon iconClassName={'sc-some-icon-class'} adjustHeight={24}>
          A text for button
        </ButtonWithCustomIcon>
      );
    });

    it('matches snapshot', () => {
      expect(mountToJson(component)).toMatchSnapshot();
    });

    it('prints no errors from propTypes validation', () => {
      expect(errorConsole).not.toHaveBeenCalled();
    });
  });
});
