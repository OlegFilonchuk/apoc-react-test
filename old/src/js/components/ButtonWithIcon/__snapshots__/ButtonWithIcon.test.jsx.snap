// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createButtonWithIcon with default set of props matches snapshot 1`] = `
<ButtonWithIcon
  adjustHeight={24}
  adjustLineHeight={undefined}
  className=""
  iconClassName="sc-some-icon-class"
  onClick={[Function]}
>
  <Button
    adjustHeight={24}
    className="button-with-icon"
    data-test-element=""
    disabled={false}
    id=""
    isLinkView={false}
    isSelected={false}
    name=""
    onClick={[Function]}
    style={
      Object {
        "lineHeight": undefined,
      }
    }
    title=""
    type="button"
    value={null}
  >
    <button
      className="sc-btn button-with-icon"
      data-test-element="undefined"
      disabled={false}
      id=""
      name=""
      onClick={[Function]}
      title=""
      type="button"
    >
      <i
        className="button-with-icon__icon sc-some-icon-class"
        style={
          Object {
            "height": undefined,
          }
        }
      />
      <span
        className="button-with-icon__text"
      >
        A text for button
      </span>
    </button>
  </Button>
</ButtonWithIcon>
`;
